baybe/campaign.py

```diff
@@ -213,6 +253,12 @@ def add_measurements(
             [self._measurements_exp, to_insert], axis=0, ignore_index=True
         )
 
+        # Update metadata
+        if self.searchspace.type in (SearchSpaceType.DISCRETE, SearchSpaceType.HYBRID):
```

Comment:
Why was this moved down? You just use `data` here which you do not seem to update previously. Or do you mean to use `to_insert` here?

Reply:
No, it's not about using `to_insert`. Rather, this seems to me the safer order of operations, i.e. updating the metadata only *after* the data has been actually recorded (written to the internal database). I was imagining cases where – for some reason – the actual storage step fails. For example, because the dataframe concatenation breaks (e.g. the columns mismatch or whatever other operation we might have forgotten to validate in the step earlier). And then you would end up with updated metadata even though the data is still the old one. Does that make sense? I see only positive consequences...

Reply:
True, but there is no particular error expected, i.e. what would I even put here? Basically, this statement hold for **any** code, i.e. you could always surround it with try-except 😄. It just felt to me that the new order is the more logical / less critical situation. But if you have a concrete suggestion, let me know – I have no ideas

Reply:
Hey @AVHopp, since this is not really changing any functionality and I would like to merge, I'll resolve this here. But #423 is still pending and if you really think something should be change here, please ping me again there (since these two PR go hand-in-hand anyway)